### PR TITLE
fix(core): display checkbox for allasset view

### PR DESCRIPTION
**Pull request by stonebuzz**
```checkbox```  from ```allasset (global)``` view is not available because

```item``` from ```allasset (global)``` view is ```null```

This PR handle this case

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #14684

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: templates/components/search/table.html.twig [hunk 1]
```diff
@@ -117,8 +117,8 @@
                         {% elseif itemtype == 'User' and not can_view_all_entities() and not has_access_to_user_entities(row['id']) %}
                         {% elseif item is instanceof('CommonDBTM') and item.maybeRecursive() and not has_access_to_entity(row['entities_id'])  %}
                         {% else %}
-
-                           {% if item is instanceof('CommonDBTM') and call(itemtype ~ '::isMassiveActionAllowed', [row['id']])  %}
+                           {# item is null from global asset view #}
+                           {% if item is null or item is instanceof('CommonDBTM') and call(itemtype ~ '::isMassiveActionAllowed', [row['id']])  %}
```
Comment (cedric-anne) [inline templates/components/search/table.html.twig:121]:
```suggestion
                           {% if itemtype == 'AllAssets' or (item is instanceof('CommonDBTM') and call(itemtype ~ '::isMassiveActionAllowed', [row['id']]))  %}
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: templates/components/search/table.html.twig (hunk 1 above)
Comment (cedric-anne) [inline templates/components/search/table.html.twig:121]: I just saw that itemtype is defined in `row['TYPE']` in global search view.
```suggestion
                           {% set row_itemtype = row['TYPE'] ?? itemtype %}
                           {% if call(row_itemtype ~ '::isMassiveActionAllowed', [row['id']]) %}
```

File: templates/components/search/table.html.twig [hunk 2]
```diff
@@ -117,8 +117,8 @@
...
                               {% set checked = session('glpimassiveactionselected')[itemtype][row['id']] ?? false %}
```
Comment (cedric-anne) [inline templates/components/search/table.html.twig:122]: Not related to initial bug, but this have to be updated too:
```suggestion
                              {% set checked = session('glpimassiveactionselected')[row_itemtype][row['id']] ?? false %}
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Comment by dzabaljauregui** [comment]
Hello people!
Thank you very much for the fix!
Keep up the good work!
Best regards,
Daniel (from Argentina).

**Pull request merged**